In [1]:
import numpy as np
import pandas as pd

# Series

In [2]:
aa = [1, 2, 3, 4]
aa

[1, 2, 3, 4]

In [3]:
ab = (5, 6, 7, 8)
ab

(5, 6, 7, 8)

In [4]:
ac = [9, 10, 11, 12]
ac

[9, 10, 11, 12]

In [5]:
ad = {'first':13, 'second':14, 'third':15, 'fourth':16}
ad

{'first': 13, 'second': 14, 'third': 15, 'fourth': 16}

In [6]:
ser1 = pd.Series(aa)
print(ser1)

0    1
1    2
2    3
3    4
dtype: int64


In [7]:
ser2 = pd.Series(ab)
print(ser2)

0    5
1    6
2    7
3    8
dtype: int64


In [8]:
ser3 = pd.Series(ac)
print(ser3)

0     9
1    10
2    11
3    12
dtype: int64


In [9]:
ser4 = pd.Series(ad)
print(ser4)

first     13
second    14
third     15
fourth    16
dtype: int64


In [10]:
ser5 = 17, 18, 19, 20
ser5

(17, 18, 19, 20)

In [11]:
ser5 = pd.Series(17, 18, 19, 20)
print(ser5)
# Would need to make it a series then assign it the literal values
# TypeError: Index(...) must be called with a collection of some kind, 18 was passed

TypeError: Index(...) must be called with a collection of some kind, 18 was passed

# DataFrames from Series 

In [12]:
df1 = pd.DataFrame(ser1)
df1

0
0  1
1  2
2  3
3  4

In [13]:
df1.insert(1, '1', ser2)
df1

0  1
0  1  5
1  2  6
2  3  7
3  4  8

In [14]:
df1.insert(2, '2', ser3)
df1

0  1   2
0  1  5   9
1  2  6  10
2  3  7  11
3  4  8  12

In [15]:
df1.insert(3, '3', ser4)
df1

0  1   2   3
0  1  5   9 NaN
1  2  6  10 NaN
2  3  7  11 NaN
3  4  8  12 NaN

In [16]:
# When I call ser 4 is a series in a dictionary with key as strings and values as integers. Compared to the other Series data used it has mix data types resulting in NaN values.

In [17]:
df1.loc['4'] = [5, 9, 12, 0]
df1

0  1   2    3
0  1  5   9  NaN
1  2  6  10  NaN
2  3  7  11  NaN
3  4  8  12  NaN
4  5  9  12  0.0

# Reading Data into a DataFrame

In [18]:
df2 = pd.read_csv('sakila_example.csv')
df2

title                                        description  \
0     ACADEMY DINOSAUR  A Epic Drama of a Feminist And a Mad Scientist...   
1     ACADEMY DINOSAUR  A Epic Drama of a Feminist And a Mad Scientist...   
2     ACADEMY DINOSAUR  A Epic Drama of a Feminist And a Mad Scientist...   
3     ACADEMY DINOSAUR  A Epic Drama of a Feminist And a Mad Scientist...   
4     ACADEMY DINOSAUR  A Epic Drama of a Feminist And a Mad Scientist...   
..                 ...                                                ...   
257  ATTRACTION NEWTON  A Astounding Panorama of a Composer And a Fris...   
258  ATTRACTION NEWTON  A Astounding Panorama of a Composer And a Fris...   
259        AUTUMN CROW  A Beautiful Tale of a Dentist And a Mad Cow wh...   
260        AUTUMN CROW  A Beautiful Tale of a Dentist And a Mad Cow wh...   
261        AUTUMN CROW  A Beautiful Tale of a Dentist And a Mad Cow wh...   

        category language        actor_name  
0    Documentary  English  PENELOPE GUINESS  
1    Documentary  English   CHRISTIAN GABLE  
2    Documentary  English     LUCILLE TRACY  
3    Documentary  English       SANDRA PECK  
4    Documentary  English       JOHNNY CAGE  
..           ...      ...               ...  
257          New  English         GARY PENN  
258          New  English  CHRISTOPHER WEST  
259        Games  English     DUSTIN TAUTOU  
260        Games  English     ANGELA HUDSON  
261        Games  English        JAMES PITT  

[262 rows x 5 columns]

In [19]:
df2.head(5)

title                                        description  \
0  ACADEMY DINOSAUR  A Epic Drama of a Feminist And a Mad Scientist...   
1  ACADEMY DINOSAUR  A Epic Drama of a Feminist And a Mad Scientist...   
2  ACADEMY DINOSAUR  A Epic Drama of a Feminist And a Mad Scientist...   
3  ACADEMY DINOSAUR  A Epic Drama of a Feminist And a Mad Scientist...   
4  ACADEMY DINOSAUR  A Epic Drama of a Feminist And a Mad Scientist...   

      category language        actor_name  
0  Documentary  English  PENELOPE GUINESS  
1  Documentary  English   CHRISTIAN GABLE  
2  Documentary  English     LUCILLE TRACY  
3  Documentary  English       SANDRA PECK  
4  Documentary  English       JOHNNY CAGE

In [20]:
df2.tail(10)

title                                        description  \
252       ATTACKS HATE  A Fast-Paced Panorama of a Technical Writer An...   
253       ATTACKS HATE  A Fast-Paced Panorama of a Technical Writer An...   
254       ATTACKS HATE  A Fast-Paced Panorama of a Technical Writer An...   
255  ATTRACTION NEWTON  A Astounding Panorama of a Composer And a Fris...   
256  ATTRACTION NEWTON  A Astounding Panorama of a Composer And a Fris...   
257  ATTRACTION NEWTON  A Astounding Panorama of a Composer And a Fris...   
258  ATTRACTION NEWTON  A Astounding Panorama of a Composer And a Fris...   
259        AUTUMN CROW  A Beautiful Tale of a Dentist And a Mad Cow wh...   
260        AUTUMN CROW  A Beautiful Tale of a Dentist And a Mad Cow wh...   
261        AUTUMN CROW  A Beautiful Tale of a Dentist And a Mad Cow wh...   

    category language        actor_name  
252   Sci-Fi  English      MILLA KEITEL  
253   Sci-Fi  English     GROUCHO DUNST  
254   Sci-Fi  English       BURT TEMPLE  
255      New  English          UMA WOOD  
256      New  English       RIP WINSLET  
257      New  English         GARY PENN  
258      New  English  CHRISTOPHER WEST  
259    Games  English     DUSTIN TAUTOU  
260    Games  English     ANGELA HUDSON  
261    Games  English        JAMES PITT

In [21]:
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 262 entries, 0 to 261
Data columns (total 5 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   title        262 non-null    object
 1   description  262 non-null    object
 2   category     262 non-null    object
 3   language     262 non-null    object
 4   actor_name   262 non-null    object
dtypes: object(5)
memory usage: 10.4+ KB


*Looking at the info results I can determine that there are 262 records with 5 columns. Each column has 262 records of non null values, which could mean there are duplication of rows. The data types are as objects for all 5 columns and that the memory ussage is 10.4 or more kilobytes.*

In [22]:
df2.describe()

title                                        description  \
count            262                                                262   
unique            46                                                 46   
top     ARABIA DOGMA  A Touching Epistle of a Madman And a Mad Cow w...   
freq              12                                                 12   

       category language    actor_name  
count       262      262           262  
unique       15        1           145  
top      Horror  English  OPRAH KILMER  
freq         55      262             8

*Looking at the describe results I can see that I'll be working with 262 records(counts) for all 5 columns, which means I would have to clean up some duplicated data but I do not think this summarization is not so useful. However, there are 46 titles and descriptions I can work with that fall into 15 caterogry and have only one language. I see that there are also 145 actors. It also give me aggregated information on the top title, description, category and actor. Finally, it gives me the frequency of the top items. However, I do not think this summarization will be very useful as I already have my top summarization that gives me the name of the items that have high occurences.*

In [23]:
df3 = pd.read_csv('northwind_example.csv')
df3

OrderID                    ProductName  UnitPrice  Quantity
0      10248                 Queso Cabrales       14.0        12
1      10248  Singaporean Hokkien Fried Mee        9.8        10
2      10248         Mozzarella di Giovanni       34.8         5
3      10249          Manjimup Dried Apples       42.4        40
4      10249                           Tofu       18.6         9
..       ...                            ...        ...       ...
495    10435         Mozzarella di Giovanni       27.8        10
496    10436         Gnocchi di nonna Alice       30.4        40
497    10436       Wimmers gute Semmelkndel       26.6        30
498    10436             Rhnbru Klosterbier        6.2        24
499    10436                      Spegesild        9.6         5

[500 rows x 4 columns]

In [24]:
df3.describe()

OrderID   UnitPrice    Quantity
count    500.000000  500.000000  500.000000
mean   10341.364000   23.371400   24.330000
std       54.442663   27.334436   18.069702
min    10248.000000    2.000000    1.000000
25%    10294.000000   10.400000   10.000000
50%    10341.000000   15.800000   20.000000
75%    10389.000000   27.800000   32.000000
max    10436.000000  210.800000  120.000000

*Looking at the describe results I see that it only returned three columns with the count of each column again shows that there is some data clean up to do as there is potentially duplicate records. Since I am dealing with unit price and quantity I am getting the mean, the standard deviation, the min, the max, and percentiles (where my data falls before the 25th percentile, where my data falls before the 50 th percentile, and where my data falls under the 75th percentile). I don't think I would have much use on the OrderID column for any of the rows that is returning mean, standard deviation, min, max, and percentiles. The summarizations here are different than for the previous example because it is pulling the data to be more statistical.*

# Coding a DataFrame

In [27]:
test_data = {'ID':[1,2,3,4,5,6,7,8], 
             'Name':['Alicia','Bob','Charlie','Dominic','Eve','Frank','Grace','Heidi'],
             'Age':[25,np.nan,22,19,31,35,np.nan,18],
             'City':['NY','LA','Chicago','Houston','Phoenix','Boston','Austin','San Diego'],
             'Score':[85,92,-1,65,78,np.nan,55,90]}
test_data             

{'ID': [1, 2, 3, 4, 5, 6, 7, 8],
 'Name': ['Alicia',
  'Bob',
  'Charlie',
  'Dominic',
  'Eve',
  'Frank',
  'Grace',
  'Heidi'],
 'Age': [25, nan, 22, 19, 31, 35, nan, 18],
 'City': ['NY',
  'LA',
  'Chicago',
  'Houston',
  'Phoenix',
  'Boston',
  'Austin',
  'San Diego'],
 'Score': [85, 92, -1, 65, 78, nan, 55, 90]}

In [31]:
df4 = pd.DataFrame(test_data)
df4

ID     Name   Age       City  Score
0   1   Alicia  25.0         NY   85.0
1   2      Bob   NaN         LA   92.0
2   3  Charlie  22.0    Chicago   -1.0
3   4  Dominic  19.0    Houston   65.0
4   5      Eve  31.0    Phoenix   78.0
5   6    Frank  35.0     Boston    NaN
6   7    Grace   NaN     Austin   55.0
7   8    Heidi  18.0  San Diego   90.0

In [34]:
df4.to_csv('test_data.csv', index=False)